src/charts/donut/donut-pct-chart-directive.js

```diff
@@ -0,0 +1,354 @@
+      controller: ['$scope',
+        function($scope) {
+
+          $scope.donutChartId = 'donutChart';
+          if ($scope.id) {
+            $scope.donutChartId = $scope.id + $scope.donutChartId;
+          }
+
+          $scope.updateAvailable = function(){
+            $scope.data.available = $scope.data.total - $scope.data.used;
+          };
+
+          if ($scope.data.available === undefined) {
+            $scope.updateAvailable();
+          }
+
+          $scope.getStatusColor = function(used, thresholds) {
+            if (!thresholds){
+              return '#0088CE';
+            } else {
+              if (used >= thresholds.error) {
+                return '#CC0000';
+              } else if (used >= thresholds.warning) {
+                return '#EC7A08';
+              } else {
+                return '#0088CE';
+              }
+            }
+          };
+
+          $scope.statusDonutColor = function(scope) {
+            var color = {
+              pattern: []
+            };
+            var percentUsed = scope.data.used / scope.data.total * 100.0;
+            color.pattern[0] = $scope.getStatusColor(percentUsed, scope.config.thresholds);
+            color.pattern[1] = '#D1D1D1';
+            return color;
+          };
+
+          var donutTooltip = function(scope) {
+            return {
+              contents: function (d) {
+                if (scope.config.tooltipFn) {
+                  var tootipHtml = '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                                        scope.config.tooltipFn(d) +
+                                   '</span>';
+                  return tootipHtml;
+                } else {
+                  return '<span class="donut-tooltip-pf" style="white-space: nowrap;">' +
+                            Math.round(d[0].ratio * 100) + '%' + ' ' + $scope.config.units + ' ' + d[0].name +
+                         '</span>';
+                }
+              }
+            };
+          };
+
+          $scope.getDonutData = function(scope) {
+            return {
+              columns: [
+                ['Used', scope.data.used],
+                ['Available', scope.data.available]
+              ],
+              type: 'donut',
+              donut: {
+                label: {
+                  show: false
+                }
+              },
+              groups: [
+                ['used', 'available']
+              ],
+              order: null
+            };
+          };
+
+          $scope.updateAll = function(scope) {
+            $scope.updateAvailable();
+            $scope.config.data = $scope.getDonutData($scope);
+            $scope.config.color = $scope.statusDonutColor($scope);
+            $scope.config.tooltip = donutTooltip(scope);
+          };
+
+          $scope.config = $.extend(true, chartsMixin.getDefaultDonutConfig(), $scope.config);
+          $scope.updateAll($scope);
+        }],
```

Comment:
Hah.  It's because of the verbose style of dependency injection (i.e. the []).  It would probably be happy if you kept the } at column 9 and put the ] on the next line at column 7 (since that would match the function declaration above).